### PR TITLE
 GR2 V1 Account Management: User Groups

### PR DESCRIPTION
**Pull request by dutt0**
## Overview/Summary
This is a new control for GR2 validation1

## This PR fixes/adds/changes/removes
This PR determines the new control requirement :
- if there is more than 1 user in the environment and 
- if so it is required to have at least two different user groups 
- of these user groups if at least one of them is being used in at least one of the environments CAPs.

### Breaking Changes
N/A

## Testing Evidence
image

## As part of this Pull Request I have

- [x] Checked for duplicate [Pull Requests](https://github.com/ssc-spc-ccoe-cei/azure-guardrails-solution-accelerator/pulls)
- [x] Associated it with relevant [GitHub Issues](https://github.com/ssc-spc-ccoe-cei/azure-guardrails-solution-accelerator/issues)
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/ssc-spc-ccoe-cei/azure-guardrails-solution-accelerator/tree/main)
- [x] Performed testing and provided evidence.
- [ ] Updated relevant and associated documentation.
- [x] Ensure PowerShell module versions have been updated (manually or with the ./tools/Update-ModuleVersions.ps1 script)

### CODE REVIEW

File: setup/IaC/modules/automationaccount.bicep [hunk 1]
```diff
@@ -316,6 +316,15 @@ resource module12 'modules' = if (newDeployment || updatePSModules) {
       }
     }
   }
+  resource module30 'modules' = if (newDeployment || updatePSModules) {
```
Comment (singhgss) [inline setup/IaC/modules/automationaccount.bicep:319]: @dutt0 module30 is already being used by Ali in his PR which is why I had used module31, think you will have to make module 32

**Review by singhgss — CHANGES_REQUESTED** [review]

File: setup/IaC/modules/automationaccount.bicep (hunk 1 above)
Comment (dutt0) [reply setup/IaC/modules/automationaccount.bicep:319]: Yes @singhgss Just kept it as a placeholder. I still need to update the FR msgs. I will resolve the conflict after merging all of your branches before I merge to main.

**Review by alalvi00 — CHANGES_REQUESTED** [review]
There are debug logs been printed in the code, is this intentional?

**Comment by dutt0** [comment]
Those Write-output code lines are intentional. However, removed the text 'DEBUG'

**Review by singhgss — APPROVED** [review]

**Review by alalvi00 — APPROVED** [review]

**Pull request merged**